model-optimizer/extensions/ops/ctc_greedy_decoder.py

```diff
@@ -40,11 +42,27 @@ def supported_attrs(self):
         ]
 
     @staticmethod
-    def ctc_greedy_decoder_infer(node: Node):
-        outn = node.out_node(0)
-        inn = node.in_node(0)
-        inn2 = node.in_node(1)
-        outn.shape = np.ones(4, dtype=np.int)
-        assert inn.shape[1] == inn2.shape[1], 'Batch for CTCGreedyDecoder should be the same in both inputs'
-        outn.shape[0] = inn.shape[1]
-        outn.shape[1] = inn.shape[0]
+    def infer(node: Node):
+        node_name = node.soft_get('name', node.id)
+        assert len(node.in_nodes()) == 2, \
+            "Incorrect number of inputs for {} node".format(node_name)
+
+        logits_shape = node.in_port(0).data.get_shape()
+        sequence_mask_shape = node.in_port(1).data.get_shape()
+
+        # check shapes of input tensors
+        assert len(logits_shape) == 3 and len(sequence_mask_shape) == 2, \
+            'Incorrect rank of some input tensor for {} node'.format(node_name)
+        assert logits_shape[1] == sequence_mask_shape[1], \
+            'Batch dimensions of input tensors must be the same for {} node'.format(node_name)
+        assert logits_shape[0] == sequence_mask_shape[0], \
+            'Time dimensions of input tensors must be the same for {} node'.format(node_name)
+
+        batch_size = logits_shape[1]
+        time_size = logits_shape[0]
+        node.out_port(0).data.set_shape(int64_array([batch_size, time_size, 1, 1]))
+
+        # This operation should be inferred in original layout
+        node['reinterp_shape'] = True
+        node['NCHW'] = True
```

Comment:
As far as I remember this attribute has no effect.

Reply:
Fixed.